openedx/core/djangoapps/course_groups/tests/test_cohorts.py

```diff
@@ -635,6 +636,30 @@ def test_add_user_to_cohort(self, mock_tracker):
             lambda: cohorts.add_user_to_cohort(first_cohort, "non_existent_username")
         )
 
+    def add_user_to_cohorts_race_condition(self):
+        course_user = UserFactory(username="Username", email="a@b.com")
+        course = modulestore().get_course(self.toy_course_key)
+        CourseEnrollment.enroll(course_user, self.toy_course_key)
+        first_cohort = CohortFactory(course_id=course.id, name="FirstCohort")
+        second_cohort = CohortFactory(course_id=course.id, name="SecondCohort")
+
+        membership1 = CohortMembership(user=course_user, course_user_group=first_cohort)
+        membership2 = CohortMembership(user=course_user, course_user_group=second_cohort)
+
+        with before_after.after('models.CohortMembership._innerRead', membership2.save):
```

Comment:
This line is incorrect, I keep getting the following error:
```
Traceback (most recent call last):
    openedx/core/djangoapps/course_groups/tests/test_cohorts.py line 649 in add_user_to_cohorts_race_condition
      with before_after.after('models.CohortMembership._innerRead', membership2.save):
    /usr/lib/python2.7/contextlib.py line 17 in __enter__
      return self.gen.next()
    /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/before_after/__init__.py line 56 in before_after
      original, _ = patcher.get_original()
    /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mock.py line 1227 in get_original
      target = self.getter()
    /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mock.py line 1414 in <lambda>
      getter = lambda: _importer(target)
    /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mock.py line 1098 in _importer
      thing = __import__(import_path)
   ImportError: No module named models
```
Testeng (@benpatterson @jzoldak), is this something you could help me get figured out? Or resident python expert @nedbat perhaps? I'm trying to get a patch-able reference to the `_innerRead` method in `CohortMembership`.